utils.go

```diff
@@ -187,3 +188,21 @@ func runCommandFull(args []string, includeStderr bool) (string, error) {
 func runCommand(args []string) (string, error) {
 	return runCommandFull(args, false)
 }
+
+// writeFile write data into specified file
+func writeFile(filePath string, data string) error {
+	// Normally dir should not be empty, one case is that cgroup subsystem
+	// is not mounted, we will get empty dir, and we want it fail here.
+	if filePath == "" {
+		return fmt.Errorf("no such file for %s", filePath)
+	}
+	if err := ioutil.WriteFile(filePath, []byte(data), 0700); err != nil {
```

Comment:
We shouldn't bury this `FileMode` here - `0700` might not be appropriate for all users so either that should be passed as a param, or something like `0600` used (or `cgroupsFileMode` or a new `const` added for this).

Reply:
yes, here is my modify, could you please help me review, @jodh-intel 
```
@@ -315,14 +315,15 @@ func copyParentCPUSet(current, parent string) error {
                return nil
        }
 
+       var cgroupsFileMode = os.FileMode(0600)
        if isEmptyString(currentCpus) {
-               if err := writeFile(filepath.Join(current, "cpuset.cpus"), string(parentCpus)); err != nil {
+               if err := writeFile(filepath.Join(current, "cpuset.cpus"), string(parentCpus), cgroupsFileMode); err != nil {
                        return err
                }
        }
 
        if isEmptyString(currentMems) {
-               if err := writeFile(filepath.Join(current, "cpuset.mems"), string(parentMems)); err != nil {
+               if err := writeFile(filepath.Join(current, "cpuset.mems"), string(parentMems), cgroupsFileMode); err != nil {
                        return err
                }
        }

```

`// writeFile write data into specified file
-func writeFile(filePath string, data string) error {
+func writeFile(filePath string, data string, fileMode os.FileMode) error {
        // Normally dir should not be empty, one case is that cgroup subsystem
        // is not mounted, we will get empty dir, and we want it fail here.
        if filePath == "" {
                return fmt.Errorf("no such file for %s", filePath)
        }
-       if err := ioutil.WriteFile(filePath, []byte(data), 0700); err != nil {
+       if err := ioutil.WriteFile(filePath, []byte(data), fileMode); err != nil {
                return fmt.Errorf("failed to write %v to %v: %v", data, filePath, err)
        }
        return nil
``
```